### ISSUE TITLE
`ValidateSchema::evaluate` is not thread safe

### ISSUE THREAD

**Issue by saiskee** [SpiderLabs/ModSecurity#2639]
**Describe the bug**

[ValidateSchema::evaluate](https://github.com/SpiderLabs/ModSecurity/blob/4127c1bf52d2b30a5c2c3e641b8085fd9a720f43/src/operators/validate_schema.cc#L46) is not thread safe. It is setting member variables of `ValidateSchema`, in this case `m_parserCtx` when this variable is used only once in the function. This causes a use-after-free error in the following scenario:
1. T1 runs `[xmlSchemaNewParserCtxt](https://github.com/SpiderLabs/ModSecurity/blob/4127c1bf52d2b30a5c2c3e641b8085fd9a720f43/src/operators/validate_schema.cc#L46)`
2. T2 runs `[xmlSchemaNewParserCtxt](https://github.com/SpiderLabs/ModSecurity/blob/4127c1bf52d2b30a5c2c3e641b8085fd9a720f43/src/operators/validate_schema.cc#L46)`
3. T1 runs `[freeXmlSchemaParser](https://github.com/SpiderLabs/ModSecurity/blob/4127c1bf52d2b30a5c2c3e641b8085fd9a720f43/src/operators/validate_schema.cc#L130)`
4. T2 runs `[freeXmlSchemaParser](https://github.com/SpiderLabs/ModSecurity/blob/4127c1bf52d2b30a5c2c3e641b8085fd9a720f43/src/operators/validate_schema.cc#L130)`, causing a use-after-free on m_parserCtx

**Expected behavior/fix**
`m_parserCtx` is no longer needed as a member, so we use a local variable instead.

**Comment by martinhsv** [comment]
Hi @saiskee ,

The problems with ValidateSchema were first identified (initially as a memory leak) way back with #2469 and addressed in a 3.1 branch: https://github.com/SpiderLabs/ModSecurity/tree/v3/dev/3.1-experimental

It hadn't gotten moved to v3/master so far -- partly simply due to low priority since this operator does not appear to be widely used. I will proceed to do so, however, just to clean it up if nothing else.

Although this item is, in essence, a duplicate, I'll leave it open for now as reminder since the original item was closed with the merge to the other branch.

**Comment by martinhsv** [comment]
Closing this duplicate now, since the fix that was originally applied to a dev branch has been merged to v3/master via #3005
